### PR TITLE
[RLlib; Offline RL] Add cloud filesystems to offline data input arguments.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

So far, output/write arguments offered the users already to define cloud filesystems (like GCS, S3, ABS) to write to. This PR proposes the same arguments for input/read processes. Theoretically, this could have been passed in by the user to the `input_read_method_kwargs` by setting up the filesystem object before building the `AlgorithmConfig`, but would be less comfortable and probably more error-prone.

This PR allows (similar to the output arguments) only GCS, S3, and ABS filesystems to be used.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -2466,6 +2470,13 @@ def offline_data(
                 unsure when to use the data or in which RL framework. The default is
                 to read column data, i.e. `False`. See also `output_write_episodes`
                 to define the output data format when recording.
+            input_filesystem: A cloud filesystem to handle access to cloud storage when
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:2473]: noiiice!!!

File: rllib/offline/offline_data.py [hunk 2]
```diff
@@ -30,6 +30,38 @@ def __init__(self, config: AlgorithmConfig):
         self.data_read_method_kwargs = (
             self.default_read_method_kwargs | config.input_read_method_kwargs
         )
+
+        # If a specific filesystem is given, set it up. Note, this could
+        # be `gcsfs` for GCS, `pyarrow` for S3 or `adlfs` for Azure Blob Storage.
+        # this filesystem is specifically needed, if a session has to be created
+        # with the cloud provider.
+        if self.filesystem:
+            if self.filesystem == "gcs":
```
Comment (sven1977) [inline rllib/offline/offline_data.py:39]: nit nit: Should we do:
```
if self.filesystem == "gcs":
   ...
elif self.filesystem is not None:
  raise ...
```

one less indentation.

**Review by sven1977 — APPROVED** [review]
Awesome PR. Offline RL is becoming very powerful in RLlib :D

File: rllib/offline/offline_data.py (hunk 2 above)
Comment (simonsays1980) [reply rllib/offline/offline_data.py:39]: Let's do it. I change it now. Great catch!

**Pull request merged**